tests/mmptest/src/TargetFrameworkDetectionTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using NUnit.Framework;
+using Xamarin.Utils;
+
+namespace Xamarin.MMP.Tests
+{
+	[TestFixture]
+	public class TargetFrameworkDetectionTests
+	{
+		TargetFramework [] XMTargetFrameworks = { TargetFramework.Xamarin_Mac_2_0_Mobile, TargetFramework.Xamarin_Mac_4_5_Full, TargetFramework.Xamarin_Mac_4_5_System };
+
+		string CreateTestExe (string tmpDir)
+		{
+			string path = Path.Combine (tmpDir, "b.exe");
+			File.WriteAllText (Path.Combine (tmpDir, "b.cs"), "public static class EntryPoint { public static void Main () {} }");
+			TI.RunAndAssert ("/Library/Frameworks/Mono.framework/Commands/mcs", string.Format ("-out:{0} {1}/b.cs", path, tmpDir), "CreateTestExe");
+			return path;
+		}
+
+		string GetTestMMPInvocation (string tmpDir, string libPath, TargetFramework targetFramework, bool correctReference = true)
+		{
+			string xmReference = correctReference ? GetXMReference (targetFramework) : GetWrongXMReference (targetFramework);
+			return string.Format ("-v -v -v -v -v --output={0} --arch=x86_64 --sdkroot /Applications/Xcode83.app --minos 10.7 {1} --sdk 10.12 --nolink -p --profile:{2} -a:{3}", tmpDir, libPath, targetFramework, xmReference);
```

Comment:
Do you really want to hardcode Xcode83.app here? `Configuration.sdk_root` should give you a more future-proof value.